// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////

syntax = "proto3";

package google.crypto.tink;

option java_package = "com.google.crypto.tink.proto";
option java_multiple_files = true;
option go_package = "github.com/google/tink/go/proto/aes_siv_go_proto";

// Tink implements RFC 5297 (https://www.rfc-editor.org/rfc/rfc5297) for
// AES-SIV, putting the SIV/Tag at the beginning of the ciphertext.
//
// While the RFC 5297 supports a list of associated datas, Tink only supports
// exactly one associated data, which corresponds to a list with one element in
// RFC 5297. An empty associated data is a list with one empty element, and not
// an empty list.

message AesSivKeyFormat {
  // Only valid value is: 64.
  uint32 key_size = 1;
  uint32 version = 2;
}

// key_type: type.googleapis.com/google.crypto.tink.AesSivKey
message AesSivKey {
  uint32 version = 1;
  // First half is AES-CTR key, second is AES-SIV.
  bytes key_value = 2;
}
